Formula/r/revanced-cli.rb

```diff
@@ -0,0 +1,27 @@
+class RevancedCli < Formula
+  desc "Command-line application to use ReVanced"
+  homepage "https://revanced.app"
+  url "https://github.com/ReVanced/revanced-cli/archive/refs/tags/v4.3.0.tar.gz"
+  sha256 "4a98678bb490aa5209ad8e68ea0c60f87fdb7d1354988c956b44cd885d77e9da"
+  license "GPL-3.0-only"
+
+  depends_on "openjdk@11" => :build
+  depends_on "openjdk"
+
+  def install
+    system "./gradlew", "clean", "shadow", "--no-daemon"
+    libexec.install "build/libs/revanced-cli-#{version}-all.jar"
+    bin.write_jar_script libexec/"revanced-cli-#{version}-all.jar", "revanced"
+  end
+
+  def caveats
+    <<~EOS
+      Some patches may require integrations such as ReVanced Integrations. Supply them with the option --merge.
+        For more information see https://github.com/ReVanced/revanced-cli/blob/main/docs/1_usage.md
+    EOS
+  end
+
+  test do
+    assert_equal "ReVanced CLI v#{version}\n", shell_output(bin/"revanced -V")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.